Colin Dismuke - React-Insta-Clone

cpdis opened the pull request:
📸 @John_Spraul 📸

cpdis committed:
Built folder structure and skeleton of JS files.

cpdis committed:
Imported data and began initial scaffolding for passing data between components. All data rendering onscreen.

cpdis committed:
Built scaffolding for search bar.

cpdis committed:
Added styling for SearchBar and refactored code to have separate components for each part of the post.

cpdis committed:
Finished MVP.

John-Spraul commented on README.md:
```diff
@@ -32,16 +32,16 @@
+[x] There are three major container components that you'll need to implement for this project: the Search Bar, the Post Container, and the Comment Section.
 - At the end of Day I there will be a single instance of the Search Bar being rendered at the top of the page, as well as a Post Container and a Comment Section for every piece of mock data in the `dummy-data.js` file.
-- The root App component of your application should import the dummy data from the `dummy-data.js` file with `import dummyData from './dummy-data';` and iterate over said data, passing each individual object as a prop to an instance of `PostContainer`.
+[x] The root App component of your application should import the dummy data from the `dummy-data.js` file with `import dummyData from './dummy-data';` and iterate over said data, passing each individual object as a prop to an instance of `PostContainer`.
```
consider keeping tack of these tasks in your PR comments instead of the readme 🤔 💭

John-Spraul commented on instagram/src/App.js:
```diff
@@ -0,0 +1,39 @@
+                             image={data.imageUrl}
+                             likes={data.likes}
+                             timestamp={data.timestamp}
+                             comments={data.comments} />
```
did you consider just passing down the whole `data` object, instead of breaking it down into separate props?

John-Spraul commented on instagram/src/components/CommentSection/Comment.js:
```diff
@@ -0,0 +1,11 @@
+const Comment = (props) => {
+    return (
+        <section className="comment">
+            <strong>{props.user}</strong> {props.text}
```
💪

John-Spraul commented on instagram/src/components/CommentSection/CommentSection.js:
```diff
@@ -0,0 +1,30 @@
+            <p><strong>{props.likes} likes</strong></p>
+            <section className="comments">
+                {props.comments.map(comment => (
+                    <div>
```
do you need the `Comment` component wrapped in a div here? And where's your 🔑 ❓ 🕵️

John-Spraul commented on instagram/src/components/PostContainer/PostContainer.js:
```diff
@@ -0,0 +1,34 @@
+            <section className="post">
+                <img src={props.image} alt="instagram" />
+            </section>
+            <section className="comment-container">
```
does this mean you have two `section`s with the className of "comment-container"?

John-Spraul commented on instagram/src/components/PostContainer/PostHeader.js:
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+
+const PostContainer = (props) => {
```
you named this component `PostContainer`, but I think you wanted to call it `PostHeader`

cpdis replied:
🤔Apparently so.

John-Spraul commented on instagram/src/components/SearchBar/SearchBar.js:
```diff
@@ -0,0 +1,31 @@
+        <section className="search-bar">
+            <section className="insta-logo">
+                <img src={ Instagram } alt="instagram logo"></img>
+                <p>|</p>
```
did you consider using a border instead of `|`?

cpdis replied:
I did, but I was running out of time and didn't have time to go back and change that.

John-Spraul commented on instagram/src/components/SearchBar/instagram.svg:
```diff
@@ -0,0 +1 @@
+<svg aria-hidden="true" data-prefix="fab" data-icon="instagram" class="svg-inline--fa fa-instagram fa-w-14" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path fill="black" d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7 ...
```
so official 😮

cpdis replied:
😎

John-Spraul reviewed (commented):
Keep up the good work!

cpdis committed:
Added lifecycle methods to App.js

cpdis committed:
Refactored CommentSection to be a pure component.

cpdis committed:
Added new comment component.

cpdis committed:
Added liking functionality.

cpdis committed:
Added box shadow around posts.

cpdis committed:
For John 🎁

cpdis committed:
Fixed comment adding issue in CommentSection.

cpdis committed:
Added search functionality.

cpdis committed:
Added highlight on focus for input areas.

cpdis committed:
Refactored App.js and created PostsPage.

cpdis committed:
Created Authetication.js.

cpdis committed:
Added local storage to comments and likes [behind by 1 😠]

cpdis committed:
Finished authetication using  Autheticate.js.

cpdis committed:
Finished Login component 🔑

cpdis committed:
Changed folder structure to make use with Netlify easier.

cpdis committed:
Removed localStorage from login and logging in locally.

cpdis commented:
[On Netlify...](https://colin-instaclone.netlify.com/)

cpdis committed:
Converted SearchBar CSS to styled components.

cpdis committed:
Converted PostContainer CSS to styled components.

cpdis committed:
Refactored all CSS into styled components.

John-Spraul reviewed (commented):
Made some notes. Looks like you're understanding the main concepts of the project, good work! 👍

cpdis committed:
Changes to styled components and search.

cpdis commented on src/components/CommentSection/AddComment.js:
```diff
@@ -1,13 +1,33 @@
+import styled from 'styled-components';
 import './comment.css';
 
+const Form = styled.div`
```
Thank you. Fixed comments and improved my understanding.

cpdis committed:
Login page styled using atlaskit styled components

I wanted to learn how to reuse styled components that have already been
created by much more compotent designers/developers. I came across
atlaskit by Atlassian the other day and figured that would be a good
place to start.
